api/TimeAddressableMediaStore.yaml

```diff
@@ -549,15 +549,23 @@ paths:
                 $ref: examples/flow-segments-get-200.json
     post:
       description: |
-        Register a new flow segment.
-        
-        The client is responsible for ensuring that the data written to the store this way obeys the following
-        restrictions:
+        Register a new flow segment, attaching the object id given to a point in the flow timeline.
+
+        The segment may use a newly-written object, or re-use an existing object from another flow.
+
+        For newly-written objects, the client is responsible for ensuring that the segment written to the store obeys the following restrictions:
         - The object id provided for a segment MUST be one which was received in a POST from /storage for this flow.
-        - The first timestamp in the actual media data written to the object MUST be located inside the time range
+        - The first media unit written to the object MUST have a timestamp located inside the time range
           that was associated with this object id in the response from the POST request to /storage for this flow.
-        - The start and end timestamps for the flow given in the submitted body MUST be the actual first and last
-          timestamps of the data that has been written into the object.
+        - All media units in the object SHOULD be used by the segment.
+        - The timestamps of each media unit in the file MUST equal its position on the Flow timeline, OR `ts_offset` MUST be set such that `media_unit_ts + ts_offset = segment_ts`
+        - The `sample_offset` SHOULD be zero.
+
+        For objects that have been re-used from other flows, the `sample_offset` and `sample_count` MAY be used to specify part of the object to use:
+        - The `range` (or `first_ts` and `last_ts`) fields indicate the new segment's position in the flow
+        - The sample associated with the start of the range (or `first_ts`) MUST be the one selected by `sample_offset`
+        - Likewise, the sample associated with the end of the range (or `last_ts`) MUST be the one selected by `sample_offset + sample_count`.
```

Comment:
Should this be `sample_offset + sample_count - 1` if referring to the last sample (associated with the end of the range)? It depends on `exclusive_last_ts` if it refers to the end of the range.

Reply:
Good point - I've extended this to call out the `exclusive_last_ts` case, which I think will make `last_ts == sample_offset+sample_count` because it's the _exclusive_ end, i.e. the point at which the next one will start?